packages/fuels-accounts/src/provider.rs

```diff
@@ -294,20 +298,18 @@ impl Provider {
             .unwrap_or(Ok(()))
     }
 
-    pub async fn dry_run<T: Transaction + Clone>(&self, tx: &T) -> Result<Vec<Receipt>> {
-        let receipts = self.client.dry_run(&tx.clone().into()).await?;
+    pub async fn dry_run<T: Transaction>(&self, mut tx: T) -> Result<Vec<Receipt>> {
+        tx.resolve_transaction()?;
```

Comment:
I was wondering if we should protect against the case where we don't know if a tx has been already resolved by introducing another type like `ResolvedTransaction` that results from this method call.
@hal3e considered it but was more leaned towards introducing less types into our workflow which already consists of builders, wrappers etc.